crates/kubelet/src/handle.rs

```diff
@@ -66,10 +163,16 @@ impl<S: Stop, R: AsyncRead + AsyncSeek + Unpin> RuntimeHandle<S, R> {
 
     /// Write all of the output from the running process into the given buffer.
     /// Returns the number of bytes written to the buffer
-    pub(crate) async fn output(&mut self, buf: &mut Vec<u8>) -> anyhow::Result<usize> {
-        let bytes_written = self.output.read_to_end(buf).await?;
-        self.output.seek(SeekFrom::Start(0)).await?;
-        Ok(bytes_written)
+    pub(crate) async fn output(
+        &mut self,
+        sender: hyper::body::Sender,
+        tail: Option<usize>,
+        follow: bool,
+    ) -> anyhow::Result<()> {
+        let mut output = self.handle.output();
+        output.seek(SeekFrom::Start(0)).await?;
+        tokio::spawn(stream_logs(output, sender, tail, follow));
```

Comment:
For my own edification, this is just going to return immediately and then any errors are passed to the user through `hyper::body::Sender`?

Reply:
Yes, it does. It doesn't handle errors very well (aside from logging them). Right now, the channel just gets dropped and then hyper closes the request. I could modify it to at least send the error to the client before that. I don't think that there is much else to be done since the response code / headers have already been sent at that point.

Reply:
Was able to handle errors when reading from AsyncRead. The only other error is when sending to channel, which probably cannot be relayed to the user at that point.